spec/integration/security_guides_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'swagger_helper'
+
+describe 'Security Guides', swagger_doc: 'v2/openapi.json' do
+  before do
+    allow_any_instance_of(Authentication).to receive(:identity_header).and_return(
+      Insights::Api::Common::IdentityHeader.new(
+        Base64.strict_encode64(
+          {
+            'identity': {
+              'org_id': '1234',
+              'type': 'User',
+              'user': {
+                'email': 'a@b.com',
+                'username': 'a@b.com',
+                'first_name': 'a',
+                'last_name': 'b',
+                'is_active': true,
+                'locale': 'en_US'
+              },
+              'internal': {
+                'org_id': '29329'
+              }
+            },
+            'entitlements':
+              {
+                'insights': {
+                  'is_entitled': true
+                }
+              }
+          }.with_indifferent_access.to_json
+        )
+      )
+    )
+    stub_rbac_permissions(Rbac::COMPLIANCE_ADMIN, Rbac::INVENTORY_HOSTS_READ)
+  end
+
+  path '/security_guides' do
+    get 'List all Security Guides' do
+      before do
+        [7, 8, 9].each do |os_version|
+          FactoryBot.create(
+            :v2_security_guide,
+            title: 'Guide to the Secure Configuration of Red Hat Enterprise ' \
+                   "Linux #{os_version}",
+            description: 'This guide presents a catalog of security-relevant ' \
+                         "configuration settings for Red Hat Enterprise Linux #{os_version}.",
+            os_major_version: os_version
+          )
+        end
+      end
+
+      tags 'security_guide'
+      description 'Lists all Security guides'
+      operationId 'ListSecurityGuides'
+      content_types
+
+      pagination_params
+      parameter name: :sort_by,
+                in: :query,
+                required: false,
+                description: 'An array of fields with an optional direction ' \
+                  '(`<key>:asc` or `<key>:desc`) to sort the results.<br>' \
+                  'If no direction is selected, `<key>:asc` is used by default.',
+                schema: {
+                  type: :array,
+                  items: { enum: sort_combinations(V2::SecurityGuide) }
+                }
```

Comment:
This should be probably a parametrized method as it will be used across endpoints.